test/python/test_deviceless_aot_compilation.py

```
@@ -101,3 +104,100 @@ def test_device_properties():
+    graph_build.build([cudnn.heur_mode.A, cudnn.heur_mode.FALLBACK])
+    blob = graph_build.serialize()
+
+    # ── sub-test 1: wrong-arch devprop must raise ────────────────────────────
+    dp_wrong = _make_wrong_arch_devprop()
+    graph_wrong = cudnn.pygraph(
+        io_data_type=cudnn.data_type.FLOAT,
+        intermediate_data_type=cudnn.data_type.FLOAT,
+        compute_data_type=cudnn.data_type.FLOAT,
+        device_property=dp_wrong,
+    )
+    with pytest.raises(Exception):
```

Ruff (0.16.1)

[warning] Do not assert blind exception: `Exception`

(B017)